core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/lru/CachedBlockQueue.java

```diff
@@ -77,9 +77,6 @@ public CachedBlockQueue(final long maxSize, final long blockSize) {
    * @return true if the CachedBlock was added to the queue
    */
   public boolean add(final CachedBlock cb) {
-    if (cb.heapSize() > maxSize) {
-      return false;
-    }
```

Comment:
I had previously added this restriction.  It seems like if the block you want to cache is bigger than the available heap size, then it shouldn't be included.  However, the code elsewhere assumes that if there is any space available in the queue, always add the block (so, it's OK if the cache extends beyond the heap size for a single block)